Lib/asyncio/tasks.py

```diff
@@ -52,19 +49,29 @@ def current_task(cls, loop=None):
 
         None is returned when called not in the context of a Task.
         """
-        if loop is None:
-            loop = events.get_event_loop()
-        return cls._current_tasks.get(loop)
+        warnings.warn("Task.current_task() is deprecated, "
+                      "use asyncio.current_task() instead",
+                      PendingDeprecationWarning,
+                      stacklevel=2)
+        try:
+            return current_task(loop)
+        except RuntimeError:
+            return None
 
     @classmethod
     def all_tasks(cls, loop=None):
         """Return a set of all tasks for an event loop.
 
         By default all tasks for the current event loop are returned.
         """
-        if loop is None:
-            loop = events.get_event_loop()
```

Comment:
You should keep `get_event_loop()` here for backwards compatibility.  So that code that calls `Task.all_tasks()` before a loop is created continues to work in 3.7 (although it will raise a warning).

`asyncio.all_tasks()` should use `get_running_loop()`.

Reply:
Not sure about `asyncio.all_tasks()` -- calling the function outside of coroutine context might make sense.
`RuntimeError` from `asyncio.current_task()` is more reasonable but I not sure again.

Reply:
> Not sure about asyncio.all_tasks() -- calling the function outside of coroutine context might make sense.

Then pass a loop object explicitly.  The problem with `get_event_loop()` is that it can create a new loop object implicitly.  Or it may fail at creating it -- that function is weird.